pages/agent/gcloud.md.erb

```diff
@@ -0,0 +1,183 @@
+2017-02-16 03:34:11 INFO   Successfully registered agent "buildkite-agent-3220167863-1cvr5" with meta-data []
+2017-02-16 03:34:11 INFO   Connecting to Buildkite...
+2017-02-16 03:34:11 INFO   Agent successfully connected
+2017-02-16 03:34:11 INFO   You can press Ctrl-C to stop the agent
+2017-02-16 03:34:11 INFO   Waiting for work...
+```
+
+To run more than one agent you can increase `replicas`:
+
+```
+$ kubectl scale --replicas=5 deployment buildkite-agent
+deployment "buildkite-agent" scaled
+
+$ kubectl get pods
+NAME                               READY     STATUS    RESTARTS   AGE
+buildkite-agent-3220167863-0m269   1/1       Running   0          24s
+buildkite-agent-3220167863-0mtps   1/1       Running   0          24s
+buildkite-agent-3220167863-1cvr5   1/1       Running   0          3m
+buildkite-agent-3220167863-9psz9   1/1       Running   0          24s
+buildkite-agent-3220167863-m3nm2   1/1       Running   0          24s
+```
+
+From here you may also want to store [an SSH key for the agent](http://buildkite.dev/docs/agent/ssh-keys) in a secret and map it into the containers for accessing private repositories:
+
+```
+# Use an existing keypair, or generate a new one with something like:
+$ ssh-keygen -t rsa -b 2048 -N "" -C builkdite-agent -f id_rsa
+Generating public/private rsa key pair.
+Your identification has been saved in id_rsa.
+Your public key has been saved in id_rsa.pub.
+...
+
+# Create a secret containing the ssh keys:
+$ kubectl create secret generic buildkite-agent-ssh \
+    --from-file id_rsa=id_rsa \
+    --from-file id_rsa.pub=id_rsa.pub
+secret "buildkite-agent-ssh" created
+
+# Change the spec to include new volumes to map the ssh key into place (because
+# this is Docker the git and ssh processes are running as root):
+$ kubectl edit deployment buildkite-agent
+...
+spec:
+  containers:
+  - ...
+    volumeMounts:
+    - name: ssh-key
+      subPath: id_rsa
+      mountPath: /root/id_rsa
+      readOnly: true
+    - name: ssh-key
+      subPath: id_rsa.pub
+      mountPath: /root/id_rsa.pub
+      readOnly: true
+    ...
+  volumes:
+  - name: ssh-key
+    secret:
+      secretName: buildkite-agent-ssh
+      defaultMode: 256
+  ...
+```
+
+The agent configuration file lives in `/buildkite/buildkite-agent.cfg` and hooks are in `/buildkite/hooks/` if you wish to map in additional volumes for customisation. Also see [our Docker setup instructions](/docs/agent/docker) for more details on configuring and customising the Buildkite Agent running in Docker.
+
+## Uploading artifacts to Google Cloud Storage
+
+You can upload [artifacts](/docs/builds/artifacts) created by your builds to your own [Google Cloud Storage](https://cloud.google.com/storage) bucket. Configure the agent to target your bucket by exporting the following environment variables using an [environment agent hook](/docs/agent/hooks) (this can not be set via the Buildkite web interface, API, or during pipeline upload):
```

Comment:
The bit about agent permissions is a little disconnected too, the second sentence isn't really connected to the first one. Like, I'm not sure that the second sentence is describing how to check that the agent has permission to create new objects.